### PR TITLE
A bunch of changes

### PR DESCRIPTION
**Pull request by Bod9001**
fixes, hydroponics Trey initial state, seed vendor contents, Mid round spawning of masp a bit of a workaround, dump the contents of Storage item sprite handler optimisation

### CODE REVIEW

File: UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs [hunk 1]
```diff
@@ -319,24 +334,56 @@ public void ServerPerformInteraction(PositionalHandApply interaction)
...
+				case PickupMode.DropClick:
+					if (canQuickEmpty)
+					{
+						// Drop all items that are inside this storage
+						var slots = itemStorage.GetItemSlots();
+						if (slots == null)
+						{
+							if (!CustomNetworkManager.Instance._isServer)
+							{
+								Chat.AddExamineMsgToClient("It's already empty!");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs]: This is server side so wont ever send, need to use Chat.AddExamineMsgFromServer

File: UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs [hunk 2]
```diff
@@ -319,24 +334,56 @@ public void ServerPerformInteraction(PositionalHandApply interaction)
...
+						if (PlayerManager.PlayerScript == null) return;
+						if (Validations.IsInReach(PlayerManager.PlayerScript.registerTile.WorldPosition ,interaction.WorldPositionTarget) == false) return;
+						if (MatrixManager.IsPassableAt( interaction.WorldPositionTarget.RoundToInt(), CustomNetworkManager.Instance._isServer) == false) return;
+
+							PlayerManager.PlayerScript.playerNetworkActions.CmdDropAllItems(itemStorage.GetIndexedItemSlot(0)
+							.ItemStorageNetID, interaction.WorldPositionTarget);
+
+						if (!CustomNetworkManager.Instance._isServer)
+						{
+							Chat.AddExamineMsgToClient($"You start dumping out the {gameObject.ExpensiveName()}.");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs]: Same here

**Comment by Bod9001** [comment]
thanks

**Comment by ThatDan123** [comment]
Also need to remove the !CustomNetworkManager.Instance._isServer checks

**Comment by Bod9001** [comment]
where?

File: UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs [hunk 3]
```diff
@@ -319,24 +334,56 @@ public void ServerPerformInteraction(PositionalHandApply interaction)
...
+
+					break;
+				case PickupMode.DropClick:
+					if (canQuickEmpty)
+					{
+						// Drop all items that are inside this storage
+						var slots = itemStorage.GetItemSlots();
+						if (slots == null)
+						{
+							if (!CustomNetworkManager.Instance._isServer)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs]: Here

File: UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs [hunk 4]
```diff
@@ -319,24 +334,56 @@ public void ServerPerformInteraction(PositionalHandApply interaction)
...
+							return;
+						}
+						if (PlayerManager.PlayerScript == null) return;
+						if (Validations.IsInReach(PlayerManager.PlayerScript.registerTile.WorldPosition ,interaction.WorldPositionTarget) == false) return;
+						if (MatrixManager.IsPassableAt( interaction.WorldPositionTarget.RoundToInt(), CustomNetworkManager.Instance._isServer) == false) return;
+
+							PlayerManager.PlayerScript.playerNetworkActions.CmdDropAllItems(itemStorage.GetIndexedItemSlot(0)
+							.ItemStorageNetID, interaction.WorldPositionTarget);
+
+						if (!CustomNetworkManager.Instance._isServer)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs]: And here

**Comment by Bod9001** [comment]
done

**Comment by Bod9001** [comment]
what do you think about the poke scene thing?

**Comment by ThatDan123** [comment]
If you want to test to see if it fixes after round start loading you'll need to add this to the player networked actions:

image

Or i guess i can PR to this branch if you want

I assume you've built tested to make sure your changes work fine and you didnt have issues? Then it might have fixed the round start issue which will be good and will mean theres a good chance it'll fix the above issue too, hopefully

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 5]
```diff
@@ -559,21 +561,28 @@ private void SetImageSprite(Sprite value)
...
-			spriteRenderer.GetPropertyBlock(block);
-			paletteOrNull = getPaletteOrNull();
-			if (paletteOrNull != null && paletteOrNull.Count == 8)
+			if (isPaletted() && PaletteSet == false)
 			{
-				List<Vector4> pal = paletteOrNull.ConvertAll((c) => new Vector4(c.r, c.g, c.b, c.a));
-				block.SetVectorArray("_ColorPalette", pal);
-				block.SetInt("_IsPaletted", 1);
+				PaletteSet = true;
+				var palette = getPaletteOrNull();
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:567]: Codacy found an issue: [Rename 'palette' which hides the field with the same name.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Messages/Server/SubScenes/PokeClientSubScene.cs  1
- UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs  6
         

Clones removed
==============
+ UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909&bid=20733023)

File: UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs [hunk 6]
```diff
@@ -244,6 +247,7 @@ private void RemoveCallbackInternal(CallbackCollection collection, Action callba
 
 	private void Update()
 	{
+		CashedDeltaTime = Time.deltaTime;
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs:250]: Codacy found an issue: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 7]
```diff
@@ -559,21 +561,28 @@ private void SetImageSprite(Sprite value)
...
+				if (palette == null && palette.Count == 8)
+				{
+					MaterialPropertyBlock block = new MaterialPropertyBlock();
+					spriteRenderer.GetPropertyBlock(block);
+					List<Vector4> pal = palette.ConvertAll<Vector4>((Color c) => new Vector4(c.r, c.g, c.b, c.a));
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:572]: Codacy found an issue: [Remove the type specification; it is redundant.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

File: UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs [hunk 8]
```diff
@@ -12,6 +12,9 @@
 /// </summary>
 public class UpdateManager : MonoBehaviour
 {
+
+	public static float CashedDeltaTime = 0;
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs:16]: Codacy found an issue: [Change the visibility of 'CashedDeltaTime' or make it 'const' or 'readonly'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 9]
```diff
@@ -559,21 +561,28 @@ private void SetImageSprite(Sprite value)
...
-			paletteOrNull = getPaletteOrNull();
-			if (paletteOrNull != null && paletteOrNull.Count == 8)
+			if (isPaletted() && PaletteSet == false)
 			{
-				List<Vector4> pal = paletteOrNull.ConvertAll((c) => new Vector4(c.r, c.g, c.b, c.a));
-				block.SetVectorArray("_ColorPalette", pal);
-				block.SetInt("_IsPaletted", 1);
+				PaletteSet = true;
+				var palette = getPaletteOrNull();
+				if (palette == null && palette.Count == 8)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:568]: Codacy found an issue: [Either reverse the equality operator in the 'palette' null test, or reverse the logical operator that follows it.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

**Comment by Bod9001** [comment]
yeah, not too sure what the issue is You are talking abou, but feel free to Pushed stuff to the branch you think it will fix it, tested with editor connecting to Build

**Comment by Bod9001** [comment]
all good now?

**Comment by ThatDan123** [comment]
uh sorry one sec, im getting git confused

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Messages/Server/SubScenes/PokeClientSubScene.cs  1
- UnityProject/Assets/Scripts/Clothing/BackPack/InteractableStorage.cs  6
         

Clones removed
==============
+ UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909&bid=20733023)

File: UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs (hunk 8 above)
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs:16]: Codacy found an issue: [Change the visibility of 'CashedDeltaTime' or make it 'const' or 'readonly'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

File: UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs (hunk 6 above)
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Managers/UpdateManager/UpdateManager.cs:250]: Codacy found an issue: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs (hunk 5 above)
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:567]: Codacy found an issue: [Rename 'palette' which hides the field with the same name.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 10]
```diff
@@ -559,21 +561,28 @@ private void SetImageSprite(Sprite value)
...
-				List<Vector4> pal = paletteOrNull.ConvertAll((c) => new Vector4(c.r, c.g, c.b, c.a));
-				block.SetVectorArray("_ColorPalette", pal);
-				block.SetInt("_IsPaletted", 1);
+				PaletteSet = true;
+				var palette = getPaletteOrNull();
+				if (palette != null && palette.Count == 8)
+				{
+					MaterialPropertyBlock block = new MaterialPropertyBlock();
+					spriteRenderer.GetPropertyBlock(block);
+					List<Vector4> pal = palette.ConvertAll<Vector4>((Color c) => new Vector4(c.r, c.g, c.b, c.a));
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:572]: Codacy found an issue: [Remove the type specification; it is redundant.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)

**Pull request merged**